src/engine/enginebufferscalest.cpp

```diff
@@ -40,12 +40,19 @@ EngineBufferScaleST::EngineBufferScaleST(ReadAheadManager *pReadAheadManager)
     m_pSoundTouch = new soundtouch::SoundTouch();
     m_pSoundTouch->setChannels(2);
     m_pSoundTouch->setRate(m_dRateOld);
-    m_pSoundTouch->setTempo(m_dTempoOld);
     m_pSoundTouch->setPitch(1.0);
     m_pSoundTouch->setSetting(SETTING_USE_QUICKSEEK, 1);
     m_pSoundTouch->setSampleRate(m_iSampleRate > 0 ? m_iSampleRate : 44100);
 
     buffer_back = new CSAMPLE[kiSoundTouchReadAheadLength*2];
+
+    // Setting the tempo to a very low value will force SoundTouch
```

Comment:
do you know this is how soundtouch works?  You are just copying my comment from the rubberband code, but I don't know that the same allocation behavior applies to soundtouch